framework/tlvf/src/include/tlvf/BaseClass.h

```diff
@@ -36,6 +36,14 @@ class BaseClass {
     bool is_finalized() { return m_finalized__; };
     void addInnerClassList(std::shared_ptr<ClassList> list) { m_inner__ = list; };
     std::shared_ptr<ClassList> getInnerClassList() { return m_inner__; };
+    template <class T> std::shared_ptr<T> class_cast()
```

Comment:
Please add documentation for new functions. In this case, it should be in framework/tlvf/README.md because that's where the rest of the documentation is (though personally I'd prefer to have everything as doxygen).

Reply:
So lets start with updating the header only, we can remove the documentation later on from the README.md since it is obsolete. @CoralMalachi can you do that as part of the update to the README.md file?